pkg/controller/common.go

```diff
@@ -78,6 +81,26 @@ func (c *Controller) GetAndFilter(ctx context.Context, key client.ObjectKey, obj
 	return true, nil
 }
 
+func (c *Controller) AreNetworkPoliciesEnabled(ctx context.Context, resource resources.Resource) (bool, error) {
+	for name, value := range resource.GetAnnotations() {
+		if name == harbormetav1.NetworkPoliciesAnnotationName {
+			return value == harbormetav1.NetworkPoliciesAnnotationEnabled, nil
+		}
+	}
+
+	defaultNetworkPoliciesStatusItem, err := configstore.Filter().Store(c.ConfigStore).Slice(config.NetworkPoliciesStatusKey).GetFirstItem()
```

Comment:
Please use helpers function to fetch configuration values.

```golang
	networkPolicyEnabled, err := c.GetString(config.NetworkPoliciesStatusKey, harbormetav1.NetworkPoliciesAnnotationDisabled)
```

Also I suggest to have boolean instead of string.

```suggestion
	networkPolicyEnabled, err := c.GetBool(config.NetworkPoliciesStatusKey, false)
```